packages/x-date-pickers/src/models/adapters.ts

```diff
@@ -0,0 +1,586 @@
+
+  // constructor (options?: { formats?: DateIOFormats, locale?: any, instance?: any });
+
+  /**
+   * Create a date in the date library format.
+   * If no `value` parameter is provided, creates a date with the current timestamp.
+   * If a `value` parameter is provided, pass it to the date library to try to parse it.
+   * @template TDate
+   * @param {any} value The optional value to parse.
+   * @returns {TDate | null} The parsed date.
+   */
+  date(value?: any): TDate | null;
+  /**
+   * Convert a date in the library format into a JavaScript `Date` object.
+   * @deprecate Will be removed in v7.
+   * @template TDate
+   * @param {TDate} value The value to convert.
+   * @returns {Date} the JavaScript date.
+   */
+  toJsDate(value: TDate): Date;
+  /**
+   * Parses an iso string into a date in the date library format.
+   * @deprecate Will be removed in v7.
+   * @template TDate
+   * @param {string} isoString The iso string to parse.
+   * @returns {TDate} the parsed date.
+   */
+  parseISO(isoString: string): TDate;
+  /**
+   * Stringifies a date in the date library format into an ISO string.
+   * @deprecate Will be removed in v7.
+   * @template TDate
+   * @param {TDate} value The date to stringify.
+   * @returns {string} the iso string representing the date.
+   */
+  toISO(value: TDate): string;
+  /**
+   * Parses a string date in a specific format.
+   * @template TDate
+   * @param {string} value The string date to parse.
+   * @param {string} format The format in which the string date is.
+   * @returns {TDate | null} The parsed date.
+   */
+  parse(value: string, format: string): TDate | null;
+  /**
+   * Returns the code of the locale currently used by the adapter.
+   * @returns {string} The code of the locale.
+   */
+  getCurrentLocaleCode(): string;
+  /**
+   * Returns `true` if the current locale is using 12 hours cycle (i.e: time with meridiem).
+   * @returns {boolean} `true` if the current locale is using 12 hours cycle.
+   */
+  is12HourCycleInCurrentLocale(): boolean;
+  /**
+   * Creates a format with no meta-token (e.g: `LLL` or `PP`).
+   * @param {string} format The format to expand.
+   * @returns {string} The expanded format.
+   */
+  expandFormat: (format: string) => string;
+  /**
+   * Returns user readable format (taking into account localized format tokens), useful to render helper text for input (e.g. placeholder).
+   * @deprecated  Will be removed in v7.
+   * @param {string} format The format to analyze.
+   * @returns {string} The helper text of the given format.
+   */
+  getFormatHelperText(format: string): string;
+  /**
+   * Returns `true` if the date is null.
+   * @template TDate
+   * @param {TDate | null} value The date to test.
+   * @returns {boolean} `true` if the date is null.
+   */
+  isNull(value: TDate | null): boolean;
+  /**
+   * Returns `true` if the date is valid.
+   * TODO v7: Type `value` to be `TDate | null` and make sure the `isValid(null)` returns `false`.
```

Comment:
I'm wondering if it's the good place to put TODO. Will they be used to generate docs, and so we will get them in the documentation. Would it be clearer as follow?

```js
// TODO: ....
/** ....*/
```

Reply:
> Will they be used to generate docs, and so we will get them in the documentation

The adapter are not documented, but even if they were there is a skip in the doc generation for the line starting with `TODO` :+1:

Reply:
I will note those `TODO` in the v7 Github issue of course once we validate them :+1:

Reply:
I was also thinking about the JSDocs helpers. But I double lot of people (except us) will use the adapters directly

![image](https://user-images.githubusercontent.com/45398769/228236388-702f9629-34fe-48be-ba39-e1b9a9a70894.png)

Reply:
Oh you are right, I did not think about those...